src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -1994,6 +1998,16 @@ def get_multisite_sync_status(self):
             raise DashboardException(error, http_status_code=500, component='rgw')
         return {}
 
+    def get_daemon_realm_name(self, daemon_name: str) -> str:
+        for _, server in CephService.get_service_map('rgw').items():
```

Comment:
Do we have any other approach to get realm name ? Rather than service map. It a lot of data to query.
I think we might be hitting this often since its  a status, or do we fetch only once, when the dashboard loads initially ? 
I expect we should have some other function or prom metric ?

Reply:
@afreen23 , I couldn't find a better and reliable daemon name to realm name mapping than service map, however there is a pre-existing method that does the same thing for us so using that now. Updated the PR.